### PR TITLE
Update: Allow custom class to grid cell component

### PR DESCRIPTION
**Pull request by ChaoLiangSuper**
## Description

- Add `addonClasses` prop to grid cell component
- Added one example using custom class to doc
- Added test of using custom class

## Motivation and Background Context

In the real case, when using `GridCell` component for a large paragraph or long text. The text will go out of the border. So adding a custom class like `width: 100%` will solve this problem. 

Moreover, providing custom class will allow customization for this component.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

1. Loading `GridCell` component and passing an array of class names
2. After render, those class names will be added on to the component.

## Screenshots (if appropriate):
screen shot 2018-02-20 at 11 51 47 am

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [ ] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: docs/examples/GridExample.jsx [hunk 1]
```diff
@@ -23,30 +23,47 @@ class GridExample extends React.PureComponent {
           <GridCell>Content</GridCell>
           <GridCell onClick={cellClicked}>This Cell logs clicks.</GridCell>
         </GridRow>
+        <GridRow>
+          <GridCell addonClasses={['full-width']}>
```
Comment (ChaoLiangSuper) [inline docs/examples/GridExample.jsx]: Better naming for this prop?

Comment (nimishjha) [reply docs/examples/GridExample.jsx]: ```customClasses``` maybe? Doesn't really matter.

Comment (omgaz) [reply docs/examples/GridExample.jsx]: For now I'd aim for consistency. I think additionalclassnames is used msot places but 'addonClassnames' may be better

File: docs/examples/styles.scss [hunk 2]
```diff
@@ -63,3 +63,7 @@
     }
   }
 }
+
+.full-width {
```
Comment (omgaz) [inline docs/examples/styles.scss]: Maybe we should start isolating these docs only styles like .docs__full-width

Comment (ChaoLiangSuper) [reply docs/examples/styles.scss]: Not sure, this class is only used in this example not even for docs. I agree with adding prefix as `doc__`, but not sure it is needed here? @omgaz

Comment (omgaz) [reply docs/examples/styles.scss]: Sure thing.

**Comment by omgaz** [comment]
Merging 🍜

**Pull request merged**